src/components/Pagination.jsx

```diff
@@ -41,75 +42,35 @@ export const Pagination = ({
         positionOLastItemOnPage
       );
     }
-  }, [activePageIDState, itemsPerPage, setCurrentPageItems, listOfItems]);
+  }, [currentPageState, itemsPerPage, setCurrentPageItems, listOfItems]);
 
   const nextButtonDisabled = useMemo(
-    () => totalPages === activePageIDState,
-    [totalPages, activePageIDState]
+    () => totalPages === currentPageState,
+    [totalPages, currentPageState]
   );
   const prevButtonDisabled = useMemo(
-    () => activePageIDState === 1,
-    [activePageIDState]
+    () => currentPageState === 1,
+    [currentPageState]
   );
 
   function onPageChange(pageNumber) {
     if (!pageNumber) {
       throw new Error('Missing number of page!');
     }
-    setActivePageIDState(pageNumber);
+    setCurrentPageState(pageNumber);
   }
```

Comment:
Function declarations get reinitialized on every rerender of the component, so every time something changes in your `Pagination` component, a new instance of the function is created. It's not a big issue here, but for larger ones you could use `useCallback` hook: https://dmitripavlutin.com/dont-overuse-react-usecallback/